PubSub/src/Subscription.php

```diff
@@ -694,17 +706,31 @@ public function pull(array $options = [])
      * }
      * ```
      *
+     * When the option `returnFailures` is set, and if a message is failed with a
+     * temporary failure code, it will be retried with an exponential delay. This will also make sure
+     * that the permanently failed message is returned to the caller. This is only true for a
+     * subscription with 'Exactly Once Delivery' enabled.
+     * Read more about EOD: https://cloud.google.com/pubsub/docs/exactly-once-delivery
+     * Example:
+     * ```
+     * $messages = $subscription->pull();
+     *
+     * foreach ($messages as $message) {
+     *     $subscription->acknowledge($message, ['returnFailures' => true]);
```

Comment:
Since there is no official suggestions on this and it's pretty much contextual, I have added a small comment that the failed messages can be logged and retried later.